### PR TITLE
Add 'ConceptFrame' based on @jman6495's window decoration concept design

### PR DESCRIPTION
**Pull request by trimental**
This PR implements reddit user jman6495's window decoration concept design as a new frame to choose from in sctk, I think the design is quite modern, minimalist and doesn't commit too much to one platforms style. 

I've asked for the designer's permission before implementing these decorations and he seems happy to help sctk by providing use of his design, hopefully this will fix a few issues about the 'ugly' decorations provided by BasicFrame, especially considering gnomes firm resistance to server side decorations in wayland.

'ConceptFrame' is just a place holder title until jman6495 responds about what his design is called or otherwise we could maybe name it something generic like 'BGRTabbedFrame' or 'MinimalistFrame', etc.

### CODE REVIEW

**Comment by Drakulix** [comment]
Maybe offer a way to change the color palette so e.g. a dark theme can be possible?

**Comment by vberger** [comment]
I must say, I quite like these decorations. :+1:

**Comment by trimental** [comment]
@Drakulix to abstract the palette over the `Frame` trait we would need to take an argument of something like `&[[u8; 4]]` or `Vec<[u8; 4]>` this might be a little abstract api wise, as an alternative an enum of color themes could be passed such as `enum FrameTheme { Light, Dark }`. If passing by argb8888 array was chosen maybe we could have frame specific structs such as `struct ConceptPalette { red_button_shade: [u8; 4], ..etc } ` that could implement a `Palette` trait containing a function that returns a color from an index?

**Comment by vberger** [comment]
I'd be ok with introducing a `Theme` trait with methods for getting colors, and add a `set_theme()` method to the `Frame` trait (and to the `Window` type) taking a `T: Theme` as argument, and with a default implementation doing nothing.

I believe a set of color like this would cover most use cases:

- primary_color
- secondary_color
- close_button_color
- maximize_button_color
- minimize_button_color

Each of them coming in two variants, "activated" or not. In case of primary/secondary colors, "activated" variants are used to draw the borders when the window is `Activated`, and the regular variants otherwise. In the case of buttons, "activated" variants are used to draw the button when hovered by the pointer, and the regular variants when not.

How does this sound?

File: src/window/concept_frame.rs [hunk 1]
```diff
@@ -0,0 +1,986 @@
...
+        Location::Button(UIButton::Minimize)
+    } else {
+        Location::Head
+    }
+}
+
+/// A minimalistic set of decorations
+///
+/// This class draws minimalistic decorations, which are arguably not very
+/// beautiful, but functional.
```
[the remark this answers is not in the document]
Comment (trimental) [reply src/window/concept_frame.rs]: Yeah I agree, these decorations are beautiful *and* functional IMHO :smiley:

**Pull request merged**